src/Codegen/Types.hack

```diff
@@ -75,6 +80,20 @@ function output_type(
     return shape('type' => Str\strip_prefix($class, 'Slack\\GraphQL\\').$suffix, 'needs_await' => $needs_await);
 }
 
+/**
+ * Get the input class for a hack type which is not a primitive.
+ */
+function get_input_class(string $hack_type): ?string {
+    $rc = new \ReflectionClass($hack_type);
```

Comment:
For https://github.com/mwildehahn/hack-graphql/pull/41/files I ended threaded through the inputs we collected but I like this more since it means we don't have to pass through a bunch of dicts here. The one thing is that we'll need to handle catching the error raised if this isn't a class, ie. I think this will look something like:

```
function get_input_type(string $hack_type): ?string {
  $rc = null;
  try {
    $rc = new \ReflectionClass($hack_type);
  catch (WhateverReflectionException) {}
  
  if ($rc is nonnull) {...}
  
  $rt = null;
  try {
    $rt = new \ReflectionType($hack_type);
  catch (WhateverReflectionException) {}
  
  if ($rt is nonnull) {...}
  
  ...
}
```

I can adjust this in my PR after you merge this

Reply:
Updating.